frontend/app/main.py

```diff
@@ -148,36 +151,19 @@ def send_static(path):
 @app.route('/leaderboard.html')
 @fresh_login_required
 def serve_leaderboard():
-    users = leaderboard.find({}, {'picture': 1, 'email': 1, 'num_of_checks': 1})
-    users = [tuple(user.values())[1:] for user in users]
-    users = sorted(users, key=lambda x: x[2], reverse=True)
+    users = leaderboard.find({}, {'all_answers': 0})
+    users = [[user['email'], user['name'], user['picture'], user['num_of_checks'], round(user['right_answers']/user['num_of_checks']*100, 1)] for user in users]
+    users = sorted(users, key=lambda x: [x[3], x[4]], reverse=True)
+    all_users_num_of_checks = [user[3] for user in users]
+    users = {user[0]: user[1:] for user in users} # {email: [name, picture, num_of_checks, agreement]}
```

Comment:
Кажется все 5 строчек можно редуцировать до одной
```suggestion
    users = {
        user['email']: [
            user['name'], user['picture'], 
            user['num_of_checks'], 
            round(user['right_answers']/user['num_of_checks']*100, 1)
        ]
        for user in leaderboard.find({}, {'picture': 1, 'email': 1, 'num_of_checks': 1})
    }
```